### PR TITLE
Implement filters info

### PR DESCRIPTION
**Pull request by Euklios**
This parses the output for `ffmpeg -filters` and creates corresponding objects.
Created analogue to the `codecs()` and `formats()` methods.

### CODE REVIEW

File: src/main/java/net/bramp/ffmpeg/info/FilterPattern.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package net.bramp.ffmpeg.info;
+
+import net.bramp.ffmpeg.probe.FFmpegStream;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.util.*;
+
+public class FilterPattern {
+    /** Indicates weather this pattern represents a source or a sink and therefore has no other options */
```
Comment (bramp) [inline src/main/java/net/bramp/ffmpeg/info/FilterPattern.java:10]: Typo s/weather/whether/g and later in this file

**Review by bramp — APPROVED** [review]
Thanks. I appreciate how much effort you are putting it to keep this consistent with the existing code styles.

File: src/main/java/net/bramp/ffmpeg/info/FilterPattern.java (hunk 1 above)
Comment (Euklios) [reply src/main/java/net/bramp/ffmpeg/info/FilterPattern.java]: Thanks for the sed pattern. Just pushed the typo fixes

**Comment by Euklios** [comment]
> Thanks. I appreciate how much effort you are putting it to keep this consistent with the existing code styles.

No worries, I'm currently maintaining four forks of this library (and many, many branches).
Sticking to the original version is about the only reason I can still keep them more or less up to date.

**Comment by bramp** [comment]
Thanks. Also, while I don't put much effort into this project, I'm happy to try and get all your pull requests in, so you don't need to manage so many external versions.

**Pull request merged**